src/Gui/Tree.cpp

```diff
@@ -2118,13 +2118,30 @@ bool TreeWidget::dropInDocument(QDropEvent* event, TargetItemInfo& targetInfo,
     infos.reserve(items.size());
     bool syncPlacement = TreeParams::getSyncPlacement();
 
+    App::AutoTransaction committer(
+        da == Qt::LinkAction ? "Link object" :
+        da == Qt::CopyAction ? "Copy object" : "Move object");
+
     // check if items can be dragged
     for (auto& v : items) {
         auto item = v.first;
         auto obj = item->object()->getObject();
         auto parentItem = item->getParentItem();
         if (parentItem) {
-            if (!parentItem->object()->canDragObjects() || !parentItem->object()->canDragObject(obj)) {
+            auto vpp = parentItem->object();
+            // We querry all the parents recursively.
+            bool allParentsOK = true;
+            auto parentItemRecursive = parentItem;
+            while (parentItemRecursive) {
+                if (!parentItemRecursive->object()->canDragObjectToTarget(obj, nullptr)) {

```

Comment:
Why is it up to the grandparent item to decide if an object can be dragged?

Reply:
For cases like :
Assembly/group/group/part